contracts/modules/Experimental/TransferManager/LTM/LockUpTransferManager.sol

```diff
@@ -658,20 +658,22 @@ contract LockUpTransferManager is LockUpTransferManagerStorage, TransferManager
 
     /**
      * @notice return the amount of tokens for a given user as per the partition
-     * @param _owner Whom token amount need to query
      * @param _partition Identifier
+     * @param _tokenHolder Whom token amount need to query
+     * @param _additionalBalance It is the `_value` that transfer during transfer/transferFrom function call
      */
-    function getTokensByPartition(address _owner, bytes32 _partition) external view returns(uint256){
-        uint256 _currentBalance = IERC20(securityToken).balanceOf(_owner);
-        if (_partition == LOCKED) {
-            return Math.min(getLockedTokenToUser(_owner), _currentBalance);
-        } else if (_partition == UNLOCKED) {
-            if (_currentBalance < getLockedTokenToUser(_owner)) {
-                return 0;
-            }
-            return _currentBalance.sub(getLockedTokenToUser(_owner));
+    function getTokensByPartition(bytes32 _partition, address _tokenHolder, uint256 _additionalBalance) external view returns(uint256){
+        uint256 currentBalance = (msg.sender == securityToken) ? (IERC20(securityToken).balanceOf(_tokenHolder)).add(_additionalBalance) : uint256(0);
```

Comment:
Curious as to why the function has to work only if the caller is the ST? (I agree that this will be when it is actually called, but just wondering why we had to enforce this).

Reply:
because it can be called from the modules as well. And if we allow the investor to call this function with any `_additionalvalue` then it may mislead then as per the output they get. That's why I think `_additionalvalue` will be in effect only when it is called by the securityToken.